rclcpp/include/rclcpp/time.hpp

```diff
@@ -36,7 +36,7 @@ class Time
   Time(int32_t seconds, uint32_t nanoseconds, rcl_time_source_type_t clock = RCL_SYSTEM_TIME);
 
   RCLCPP_PUBLIC
-  explicit Time(uint64_t nanoseconds, rcl_time_source_type_t clock = RCL_SYSTEM_TIME);
+  explicit Time(uint64_t nanoseconds = 0, rcl_time_source_type_t clock = RCL_SYSTEM_TIME);
```

Comment:
if this constructor has a default value, shouldn't the upper constructor `Time(int32_t seconds, uint32_t nanoseconds)` have default values as well?

Reply:
No, if two constructors can be called without arguments it wouldn't be ambiguous which one the user wants to call.

Reply:
@Karsten1987 I don't think any number of the other constructor arguments should be optional. That is just a convenience constructor anyways, which I believe calls the nanosecond version of the constructor. Also, as @dirk-thomas said, we can't have two constructors that take no arguments.